sycl/source/detail/graph_impl.hpp

```diff
@@ -459,6 +483,30 @@ class node_impl {
   }
 };
 
+class partition {
+public:
+  /// Constructor.
+  partition() : MSchedule(), MPiCommandBuffers() {}
+
+  /// List of root nodes.
+  std::set<std::weak_ptr<node_impl>, std::owner_less<std::weak_ptr<node_impl>>>
+      MRoots;
+  /// Execution schedule of nodes in the graph.
+  std::list<std::shared_ptr<node_impl>> MSchedule;
+  /// Map of devices to command buffers.
+  std::unordered_map<sycl::device, sycl::detail::pi::PiExtCommandBuffer>
+      MPiCommandBuffers;
+
+  /// @return True if the partition contains a host task
+  bool isHostTask() {
```

Comment:
Not sure if this can be const
```suggestion
  bool isHostTask() const {
```